# 07 - Business Insights & Recommendations

Synthesizes findings from EDA, feature engineering, and modeling into an executive-ready set of insights and recommended actions.

In [1]:
import sys
sys.path.insert(0, '../src')
import pandas as pd
df = pd.read_csv('../dataset/customer_churn_features.csv')
churn_rate = lambda s: (s=='Yes').mean()*100

## Headline numbers

In [2]:
overall = churn_rate(df['Churn'])
rev_at_risk = df.loc[df.Churn=='Yes','MonthlyCharges'].sum()
clv_at_risk = df.loc[df.Churn=='Yes','CustomerLifetimeValue'].sum()
print(f'Overall churn rate: {overall:.1f}%')
print(f'Customers churned: {(df.Churn=="Yes").sum():,} of {len(df):,}')
print(f'Monthly recurring revenue at risk: ${rev_at_risk:,.0f}')
print(f'Total customer lifetime value at risk: ${clv_at_risk:,.0f}')

Overall churn rate: 31.7%
Customers churned: 2,381 of 7,500
Monthly recurring revenue at risk: $205,091
Total customer lifetime value at risk: $8,792,436


## Top churn drivers, ranked by effect size

| Driver | Comparison | Churn Rate |
|---|---|---|
| Contract type | Month-to-Month vs Two Year | 44.3% vs 11.7% |
| Internet service | Fiber Optic vs DSL | 46.3% vs 22.2% |
| Complaints | 3 complaints vs 0 | 63.4% vs 25.0% |
| Tenure | 0-6 months vs 49+ months | 51.2% vs 19.5% |
| Payment method | Electronic Check vs Auto-pay | 37.2% vs ~28.5% |
| Tech support | No vs Yes | 37.0% vs 32.9% |
| Referral status | No vs Yes | 33.4% vs 27.2% |

## Recommended actions (prioritized by expected impact / effort)

1. **Convert month-to-month customers to annual contracts** via a modest discount or perk — the single largest churn-rate gap in the data (32.6 points between month-to-month and two-year plans).
2. **Investigate the Fiber Optic experience** — price, reliability, or support quality on this tier is driving churn nearly double DSL's rate despite being the premium product.
3. **Trigger a save intervention at the 2nd complaint**, before churn risk crosses 50% at the 3rd — turn support logs into an early-warning system rather than a pure cost center.
4. **Front-load onboarding support in months 0-6**, where churn risk is highest (51.2%) — proactive check-ins, not just reactive support.
5. **Nudge Electronic Check payers toward autopay** with a small incentive — low-cost, directly tied to a 9-10 point churn-rate gap.
6. **Bundle tech support / online security into new sign-ups**, even as a free trial — both add-ons show a consistent, moderate churn-reduction effect.
7. **Expand the referral program** — referred customers churn 6.2 points less, meaning referral marketing spend also pays a retention dividend, not just an acquisition one.

## How the ML model complements these rules

The rule-based segments above are useful immediately, but they treat each driver independently. The trained Logistic Regression model (ROC-AUC 0.78) combines all drivers simultaneously into a single churn probability per customer, letting the retention team rank the entire customer base by risk and target the highest-value, highest-risk accounts first — rather than acting on one rule at a time.

## Suggested KPIs to track post-implementation

- Overall churn rate (target: reduce from 31.7% toward industry benchmark ~20-24%)
- Month-to-month share of the customer base
- Complaint-to-churn conversion rate at each complaint count
- Fiber Optic segment churn rate specifically
- Model-flagged high-risk customers retained after intervention (save rate)